lib/token.cpp

```diff
@@ -35,8 +35,12 @@
 
 const std::list<ValueFlow::Value> Token::emptyValueList;
 
-Token::Token(Token **tokens) :
-    tokensBack(tokens),
+Token::Token() : Token(TokensFrontBack{0, 0}) { }
+
+Token::Token(TokensFrontBack listEnds) : Token(&listEnds) {}
```

Comment:
this looks dangerous. You are passing pointer to temporary `listEnds` parameter. It seems to me this constructor should be removed. I am also not sure if we allow delegating constructors yet, it has to be supported by gcc 4.6 etc..

Reply:
That would be in aligned with the analysis posted here: https://github.com/danmar/cppcheck/pull/1238#issuecomment-391100643